python/cudf/cudf/utils/metadata/orc_column_statistics_pb2.py

```diff
@@ -0,0 +1,549 @@
+# flake8: noqa
+# fmt: off
+# -*- coding: utf-8 -*-
+# Generated by the protocol buffer compiler.  DO NOT EDIT!
+# source: orc_column_statistics.proto
+"""Generated protocol buffer code."""
+from google.protobuf import descriptor as _descriptor
+from google.protobuf import message as _message
+from google.protobuf import reflection as _reflection
+from google.protobuf import symbol_database as _symbol_database
```

Comment:
I know this file is generated, but isn't this a new dependency that we'd need to add to the `cudf` conda recipe / conda environment files?

Reply:
Interestingly, it seems like protobuf is included here [1]. However, I added it to `cudf/conda/environments/cudf_dev_cuda*.yml` although I'm not sure - is protobuf already an added dependency and if not should I be adding it to these conda environment files?

[1] https://github.com/rapidsai/integration/blob/branch-0.16/conda/recipes/rapids-build-env/meta.yaml

Reply:
Okay, should we remove this file and add it to the gitignore then add the generation in the `setup.py`? This way if someone modifies the `.proto` file we don't need to check that the generated python file is similarly updated.

Reply:
Also need to add `protobuf` to the `cudf` conda recipe if we go this route, which I think we should.

Reply:
I'm a little unsure how to go about this. To ensure `protoc` is installed, does `setup.py` need to have something like this - http://web.mit.edu/andersk/src/protobuf-2.5.0/python/setup.py?

Reply:
`versioneer.py` is vendored in, you shouldn't need to touch that file. For `setup.py` you can add `protobuf` to https://github.com/rapidsai/cudf/blob/branch-0.16/python/cudf/setup.py#L88 as well as the `cudf` conda recipe in the `build` section: https://github.com/rapidsai/cudf/blob/branch-0.16/conda/recipes/cudf/meta.yaml#L22-L31. Then just add the logic to run the `protoc` command in the `setup.py`.

Reply:
> For setup.py you can add protobuf to https://github.com/rapidsai/cudf/blob/branch-0.16/python/cudf/setup.py#L88 as well as the cudf conda recipe in the build section: https://github.com/rapidsai/cudf/blob/branch-0.16/conda/recipes/cudf/meta.yaml#L22-L31.

This makes sense to me.

> Then just add the logic to run the protoc command in the setup.py.

How do I do this? It seems like I need to modify `cmdclass` [1] to add custom steps to build process. Instead, are you saying I can just add the logic to run `protoc` at bottom (or top?) of `setup.py`?

[1] https://github.com/rapidsai/cudf/blob/branch-0.16/python/cudf/setup.py#L100

Reply:
Presumably you could put it in the path of when a user calls `setup.py install` or `setup.py develop`, but I'm not quite sure as to how that works.

Reply:
I think there are 2 ways to do add `.proto` compilation to build process-
1. Subclass an existing command like `build_ext` or `install` and map to the new subclassed commands
2. Create a new command like `setup.py build_pb`

I'm currently trying both of these but am running into issues. But either way, it's going to require changes to build process in rapids-compose and elsewhere - right? if not, it seems like it will introduce complexity in setup.py with subclassing. Should I still keep trying to integrate this into `setup.py`?

Reply:
Alright, I've added the protobuf compilation to `setup.py` and it seems to build smoothly. Let me know what you think.